include/dsn/dist/replication/replication_app_base.h

```diff
@@ -235,6 +235,9 @@ class replication_app_base : public replica_base
     // Thread-safe.
     virtual void set_partition_version(int32_t partition_version){};
 
+    // restore the write request some info to string, it may need overload
+    virtual std::string restore_write_request(dsn::message_ex *request) { return "write request"; };
```

Comment:
I think "restore" is not a good naming here, it sounds like a "rollback", would some other names like "dump", "unmarshall" be better?